packages/browser/src/plugins/in-app-plugin/index.ts

```diff
@@ -2,208 +2,240 @@ import { Analytics } from '../../core/analytics'
-                    action: params.action,
-                    name: params.name,
-                    actionName: params.name,
-                    actionValue: params.action,
-                    message: {
-                        dismiss: function() {
-                            Gist.dismissMessage(params.message.instanceId);
-                        }
-                    }
-                }));
-            }
-            if (params.action == 'gist://close') {
-                return;
-            }
-            if (typeof deliveryId != 'undefined' && deliveryId != '') {
-                _analytics.track(JourneysEvents.Metric, {
-                    'deliveryId': deliveryId,
-                    'metric': JourneysEvents.Clicked,
-                    'actionName': params.name,
-                    'actionValue': params.action,
-                });
-                return;
-            }
-            const broadcastId:Number = params?.message?.properties?.gist?.broadcast?.broadcastIdInt;
-            if (broadcastId) {
-                const templateId:Number = params?.message?.properties?.gist?.broadcast?.templateId;
-                _analytics.track(JourneysEvents.Content, {
-                    'actionType': JourneysEvents.ClickedContent,
-                    'contentId': broadcastId,
-                    'templateId': templateId,
-                    'contentType': ContentType,
-                    'actionName': params.name,
-                    'actionValue': params.action,
-                });
-            }
-        });
-
-        Gist.events.on('eventDispatched', (gistEvent: any) => {
-            if(gistEvent.name == 'analytics:track') {
-                const trackEventName:string = gistEvent.payload?.event;
-                if(typeof trackEventName === 'undefined' || trackEventName == '') {
-                    return;
-                }
-                _analytics.track(trackEventName, gistEvent.payload?.properties, gistEvent.payload?.options);
-            }
-        });
+  async function setAnonymousId() {
+    const anonymousId = _analytics.user().anonymousId()
+    if (anonymousId) {
+      await Gist.setCustomAttribute('cio_anonymous_id', anonymousId)
     }
-
-    async function page(ctx: Context): Promise<Context> {
-        if(!_pluginLoaded)
-            return ctx;
-
-        const page:string = ctx.event?.properties?.name ?? ctx.event?.properties?.url;
-        if(typeof page === 'string' && page.length > 0) {
-            Gist.setCurrentRoute(page);
-        }
-
-        return ctx;
+  }
+
+  function attachListeners() {
+    if (!_gistLoaded || _pluginLoaded) return
+
+    _analytics.on('reset', reset)
+
+    if (settings.events) {
+      allEvents.forEach((event) => {
+        _eventTarget.addEventListener(
+          event,
+          settings?.events as EventListenerOrEventListenerObject
+        )
+      })
+      ;['messageDismissed', 'messageError'].forEach((event) => {
```

Comment:
Odd spot for a semicolon 🤔 should it be this?
```suggestion
      });
      ['messageDismissed', 'messageError'].forEach((event) => {
```

Reply:
lol weird 😄